### PR TITLE
add Imviz_RadialPlot_Example.ipynb

### PR DESCRIPTION
**Pull request by eteq**
This adds a concept notebook that shows how one can make a radial plot from a selection in imviz.

This is basically a concept for an imviz plugin.

One thing this does that is not idea: it creates a *new* bqplot scatter plot every time a new subset is created.  There's probably someway to re-use the plot and just change the data, but I couldn't figure out the right way to do this. The main problem is that I couldn't figure out a clean way to let a new data object be a different *size* from the previous one without re-creating the plot (and data object). @astrofrog  or @maartenbreddels may have ideas here?

[🐱](https://jira.stsci.edu/browse/JDAT-2055)

### CODE REVIEW

File: notebooks/concepts/Imviz_RadialPlot_Example.ipynb [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+   "execution_count": null,
+   "id": "cebe4a5a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We can use the glue_app stuff here, but I just want to\n",
+    "# see if we get the same result in Matplotlib.\n",
+    "# Should look the same as \"Subset based selection\" below.\n",
+    "#\n",
+    "# Kyle, this is what will get sent to the frontend:\n",
```
Comment (pllim) [inline notebooks/concepts/Imviz_RadialPlot_Example.ipynb:339]: @kecnry , I think this is what we need to send to bqplot: `r.ravel(), img` (two 1D float arrays).

Maybe if you read the whole "From simple aperture photometry plugin" here, it would be clearer what I mean?

With the `glue_app` thingy that Erik originally implemented here, the radial plot is interactive. But adding data to `glue_app` is not ideal because these are throw-away buffers. We don't want to keep adding/deleting them. If there is a way to make your bqplot implementation interactive without glue_app, then great. If not, maybe it is not a big deal?

File: notebooks/concepts/Imviz_RadialPlot_Example.ipynb [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+    "glue_app.add_data(plotdat_sub=dict(rdat=r, dat=subset0['SCI,1']))\n",
+    "scat2d = glue_app.scatter2d(data='plotdat_sub', x='rdat', y='dat')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9c27908d",
+   "metadata": {},
+   "source": [
+    "# Live update from subset version "
```
Comment (pllim) [inline notebooks/concepts/Imviz_RadialPlot_Example.ipynb:450]: I don't know how I feel about this. Since we are attaching the radial plot to "simple aperture photometry" plugin, having live update would make aperture photometry calculations more complicated too. Maybe this is out of scope, @PatrickOgle, @orifox, or @camipacifici?

File: notebooks/concepts/Imviz_RadialPlot_Example.ipynb (hunk 1 above)
Comment (kecnry) [reply notebooks/concepts/Imviz_RadialPlot_Example.ipynb:339]: Ok, I'll see if I can get bqplot in without your buffer concerns, and if not, we can proceed with mpl for now and re-consider down the road if we need interactivity or want more advanced functionality (adding to the main viewer, etc).

Comment (pllim) [reply notebooks/concepts/Imviz_RadialPlot_Example.ipynb:339]: In Ginga, the plot is static but you can adjust some settings and replot. For instance, the user would want to adjust the max of X-axis (zoom in/out horizontally) but there are several ways to accomplish that depending on your findings.

Comment (pllim) [reply notebooks/concepts/Imviz_RadialPlot_Example.ipynb:339]: (Actually recently Eric Jeschke might have added interactivity via key bindings but that is another story.)

**Comment by pllim** [comment]
Superseded by #1030

**Comment by pllim** [comment]
@eteq , I see that you pushed a new commit but I don't think there is any plan to merge this.

**Comment by eteq** [comment]
@pllim - yeah sorry I forgot this PR was even open! I was pushing it up to show someone else this capability.

I think there's a question, though, whether when I get a chance I should turn it into an example for a user of how they would make a "notebook plugin"? I.e.,  the point is not so much the radial plot per se, but how you go about adding UI in a notebook. (Should definitely be updated to clearly say "use the feature from #1030 instead of this notebook for the actual problem at hand" though.

**Comment by pllim** [comment]
I don't think we should provide *this* example when we're writing a plugin for it. I think that is too confusing. "Write your own plugin" tutorial should feature a really niche usage that we'll never support.

**Comment by maartenbreddels** [comment]
Happy to take a look at the bqplot issue if #1030 will use that.

**Comment by pllim** [comment]
@eteq , this PR is closed without merge. Will be happy to work with you for another example on "how to write plugin" if you can open a JIRA ticket to get it queued (if it isn't in queue already). Thanks!
